harness/determined/common/experimental/determined.py

```diff
@@ -404,3 +404,157 @@ def stream_trials_validation_metrics(
             trial_ids: List of trial IDs to get metrics for.
         """
         return trial._stream_validation_metrics(self._session, trial_ids)
+
+    def get_workspace_ids(
+        self,
+        session: api.Session,
+        workspace_names: List[str],
+    ) -> List[int]:
+        wnSet = set(workspace_names)
+        workspace_ids = []
+
+        def get_with_offset(offset: int) -> bindings.v1GetWorkspacesResponse:
+            return bindings.get_GetWorkspaces(
+                session=self._session,
+                offset=offset,
+            )
+
+        resps = api.read_paginated(get_with_offset)
+        workspaces = [w for r in resps for w in r.workspaces]
+
+        for w in workspaces:
+            if w.name in wnSet:
+                if w.archived:
+                    raise Exception(f'Workspace "{w.name}" is archived.')
+                workspace_ids.append(w.id)
+        return workspace_ids
+
+    def bind_rps_to_workspaces(
+        self,
+        resource_pool_names: List[str],
+        workspace_names: List[str],
+    ) -> None:
+        """
+        Bind resource pools to workspaces.
+
+        Arguments:
+            resource_pool_names (list(str)): The names of the resouce pools to be
+                bound.
+            workspace_names (list(str)): The names of the workspaces to be bound.
+        """
+        workspace_ids = self.get_workspace_ids(self._session, workspace_names)
+
+        for resource_pool_name in resource_pool_names:
```

Comment:
One of the principles we're reinforcing in the SDK is to keep the interface simple, and not do work that our users can easily do themselves. Here, that means
```python
def bind_rp_to_workspace(self, resource_pool_name: str, workspace_name: List[str]) -> None:
```
rather than
```python
def bind_rps_to_workspace(self, resource_pool_names: List[str], workspace_name: List[str]) -> None:
```

I'd ideally also like it even more if we could even better
```python
def bind_rp_to_workspace(self, resource_pool_name: str, workspace_name: str) -> None:
```
but I get it if product has latency targets for a use case this wouldn't perform well for.